src/components/visual-query-builder/BQWhereRow.tsx

```diff
@@ -41,6 +41,11 @@ export function BQWhereRow({ query, apiClient, onQueryChange }: BQWhereRowProps)
     [sql]
   );
 
+  // Need to keep the state in sync with the props
+  useDeepCompareEffect(() => {
+    setSql(query.sql);
+  }, [query.sql]);
+
   return (
```

Comment:
Let's skip the debounced query at all, since we have the run query button now.

Reply:
I've removed the debounce now and it works as expected.